### PR TITLE
etcd: fix proxy

### PR DESCRIPTION
**Pull request by xiang90**
1. move proxy datadir to /proxy subdir.
2. delay update proxy's cluster after validation.

Fix https://github.com/coreos/etcd/issues/2206

### CODE REVIEW

File: etcdmain/etcd.go [hunk 1]
```diff
@@ -250,19 +251,22 @@ func startProxy(cfg *config) error {
...
-		old := cls.PeerURLs()
-		cls, err = etcdserver.GetClusterFromPeers(peerURLs, tr)
+		gcls, err := etcdserver.GetClusterFromPeers(peerURLs, tr)
+		// TODO: remove the 2nd check when we fix GetClusterFromPeers
 		if err != nil {
 			log.Printf("proxy: %v", err)
 			return []string{}
 		}
+		if len(gcls.Members()) == 0 {
+			return cls.PeerURLs()
```
Comment (barakmich) [inline etcdmain/etcd.go]: We return cls.ClientURLs() below. I assume this should too

Comment (xiang90) [inline etcdmain/etcd.go]: right. my mistake.

Comment (xiang90) [inline etcdmain/etcd.go]: fixed

File: etcdmain/etcd.go [hunk 2]
```diff
@@ -250,19 +251,22 @@ func startProxy(cfg *config) error {
...
+		}
+		cls = gcls
 
 		urls := struct{ PeerURLs []string }{cls.PeerURLs()}
```
Comment (barakmich) [inline etcdmain/etcd.go]: Is this used in the following gode?

Comment (barakmich) [inline etcdmain/etcd.go]: NM

Comment (xiang90) [inline etcdmain/etcd.go]: yes. the line right after it.

**Comment by barakmich** [comment]
LGTM

**Pull request merged**